src/auth/auth.controller.ts

```diff
@@ -86,16 +86,34 @@ export class AuthController extends BaseController implements IAuthController {
 	}
 
 	async forgotPassword(req: Request, res: Response, next: NextFunction): Promise<void> {
-		this.ok(res, {});
+		try {
+			const { email } = req.body;
+			const user = await this.usersService.getUserByEmail(email);
+
+			await this.authService.sendPasswordRestorationEmail(user);
+
+			this.ok(res, `Restoration email for ${email} was sent.`);
+		} catch (e) {
+			return next(e);
+		}
 	}
 
 	async restorePassword(req: Request, res: Response, next: NextFunction): Promise<void> {
-		this.ok(res, {});
+		const user = req.user;
+		const { newPassword } = req.body;
+
+		try {
+			await this.authService.restorePassword(user, newPassword);
```

Comment:
please provide only the necessary information for the service
```js
await this.authService.restorePassword(user.id, newPassword);
```

Reply:
Done